renderdoc/driver/vulkan/vk_acceleration_structure.cpp

```diff
@@ -0,0 +1,548 @@
+      if(!SerialiseBLAS(blasAddr, results))
+        return false;
+    }
+  }
+
+  return true;
+}
+
+template <typename SerialiserType>
+bool VulkanAccelerationStructureManager::Serialise(SerialiserType &ser, ResourceId id,
+                                                   const VkInitialContents *initial,
+                                                   CaptureState state)
+{
+  if(ser.VersionLess(version_2024_0))
+  {
+    RDCERR(
+        "This capture was made with RenderDoc for Arm GPUs 2024.0, please use that to open this "
+        "capture");
+    return false;
+  }
+
+  SERIALISE_ELEMENT_LOCAL(tag, initial->tag);
+  rdcliteral name =
+      tag == VkInitialContents::SerialisedBottomLevelAccelerationStructure ? "BLAS"_lit : "TLAS"_lit;
+
+  VkDevice d = !IsStructuredExporting(state) ? m_Parent->GetDev() : VK_NULL_HANDLE;
+  const bool replayingAndReading = ser.IsReading() && IsReplayMode(state);
+  VkResult vkr = VK_SUCCESS;
+
+  byte *contents = NULL;
+  uint64_t contentsSize = initial ? initial->mem.size : 0;
+  MemoryAllocation mappedMem;
+
+  // Serialise this separately so that it can be used on reading to prepare the upload memory
+  SERIALISE_ELEMENT(contentsSize);
+
+  const VkDeviceSize nonCoherentAtomSize = m_Parent->GetDeviceProps().limits.nonCoherentAtomSize;
+
+  // the memory/buffer that we allocated on read, to upload the initial contents.
+  MemoryAllocation uploadMemory;
+  VkBuffer uploadBuf = VK_NULL_HANDLE;
+
+  if(ser.IsWriting())
+  {
+    if(initial && initial->mem.mem != VK_NULL_HANDLE)
+    {
+      const VkDeviceSize size = AlignUp(initial->mem.size, nonCoherentAtomSize);
+
+      mappedMem = initial->mem;
+      vkr = ObjDisp(d)->MapMemory(Unwrap(d), Unwrap(mappedMem.mem), initial->mem.offs, size, 0,
+                                  (void **)&contents);
+      m_Parent->CheckVkResult(vkr);
+
+      // invalidate the cpu cache for this memory range to avoid reading stale data
+      const VkMappedMemoryRange range{
+          VK_STRUCTURE_TYPE_MAPPED_MEMORY_RANGE, NULL, Unwrap(mappedMem.mem), mappedMem.offs, size,
+      };
+
+      vkr = ObjDisp(d)->InvalidateMappedMemoryRanges(Unwrap(d), 1, &range);
+      m_Parent->CheckVkResult(vkr);
+    }
+  }
+  else if(IsReplayMode(state) && !ser.IsErrored())
+  {
+    // create a buffer with memory attached, which we will fill with the initial contents
+    const VkBufferCreateInfo bufInfo{VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO, NULL, 0, contentsSize,
+                                     VK_BUFFER_USAGE_TRANSFER_SRC_BIT |
+                                         VK_BUFFER_USAGE_TRANSFER_DST_BIT |
+                                         VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT};
+
+    vkr = m_Parent->vkCreateBuffer(d, &bufInfo, NULL, &uploadBuf);
+    m_Parent->CheckVkResult(vkr);
+
+    uploadMemory = m_Parent->AllocateMemoryForResource(uploadBuf, MemoryScope::InitialContents,
+                                                       MemoryType::Upload);
```

Comment:
`AllocateMemoryForResource` does not currently set the right bits for BDA buffers to be bound to it.

Reply:
It will be in a subsequent PR (https://github.com/cmannett85-arm/renderdoc/blob/as_manager_2/renderdoc/driver/vulkan/vk_memory.cpp#L331).  This could be too heavy handed, but we'll cross that bridge when I PR it.